packages/@sanity/vision/src/components/QueryTypesContainer.tsx

```diff
@@ -0,0 +1,116 @@
+import {Box, Code, Tab, TabList} from '@sanity/ui'
+import {type TypeNode} from 'groq-js'
+import {useState} from 'react'
+import {type TFunction} from 'sanity'
+import styled from 'styled-components'
+
+export const QueryTypesWrapper = styled(Box)`
+  height: 100%;
+  width: 30%;
+  min-width: 300px;
+  @media screen and (max-width: 700px) {
+    display: none !important;
+  }
+`
+
+export const TypesContainer = styled(Box)`
+  height: calc(100% - 60px);
+  width: 100%;
+  overflow: auto;
+`
+
+export const LangContainer = styled(Box)`
+  height: 40px;
+  width: 100%;
+  overflow: auto;
+`
+
+export function QueryTypesContainer(props: {
+  typeNode?: TypeNode
+  t: TFunction<'vision', undefined>
+}) {
+  const [lang, setLang] = useState<'typescript' | 'swift' | 'go'>('typescript')
+  if (!props.typeNode) {
+    return null
+  }
+
+  return (
+    <QueryTypesWrapper overflow="auto" paddingY={3}>
+      <LangContainer>
+        <TabList space={2}>
+          <Tab
+            aria-controls="typescript-lang"
+            id="content-tab"
+            label={props.t('TypeScript')}
+            onClick={() => setLang('typescript')}
+            selected={lang === 'typescript'}
+          />
+          <></>
+        </TabList>
+      </LangContainer>
```

Comment:
I see where this is going (and this is great), but until we have more than one language, this looks kind of out of place. Maybe we can just hide this for now?

Reply:
Sure. It was mostly because I liked the heading, as it gave some context of the view. Ie your query is not matching your schema it might just say `nulm`